### PR TITLE
Social Previews: Upgrade Nudge in Modal

### LINKED ISSUE #16729: Social Previews: Implement upgrade nudge design
After #16704 gets merged, let's implement a proper design for the upgrade nudge. There is a place in the code where it fits: https://github.com/Automattic/jetpack/pull/16704/files#diff-28c6ffab5a16e5b708c41172442f041dR70

I recon the best way might be to introduce a new component `SocialPreviewsNudge` or similar so we don't complicate the modal component more. 

Design:

Figma design source:

https://www.figma.com/file/JBY5Fueh4nHqkrfMrKIhs8/Social-Components?node-id=9%3A6101 (artboard 02B)

### PR DESCRIPTION
**Pull request by marekhrabe**
Fixes #16729

#### Changes proposed in this Pull Request:
* implements the nudge design from #16729 

#### Jetpack product discussion
—

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- follow setup steps from #16615 
- this PR currently shows both paid and unpaid versions of the sidebar. Click the one with "Learn More" button and it opens up this upgrade nudge

#### Proposed changelog entry for your changes:
none

### CODE REVIEW

File: extensions/blocks/social-previews/editor.js [hunk 1]
```diff
@@ -19,14 +19,13 @@ registerJetpackPlugin( name, settings );
  * extension availability so will not render the Plugin if the extension is not
  * availabile.
  */
-const extensionAvailableOnPlan = getJetpackExtensionAvailability( 'social-previews' );
+// TODO: remove before submitting PR for a review.
```
Comment (cpapazoglou) [inline extensions/blocks/social-previews/editor.js:22]: It seems the false here should be removed (after testing & before merging)?

File: extensions/blocks/social-previews/editor.scss [hunk 2]
```diff
@@ -12,3 +12,75 @@
...
+	font-size: 1.9em;
+	line-height: 1.15;
+}
+
+.jetpack-social-previews__feature-list {
+	list-style: none;
+	margin-bottom: 2em;
+	padding-left: 1em;
+
+	li {
```
Comment (cpapazoglou) [inline extensions/blocks/social-previews/editor.scss:30]: I would suggest adding 
```css
    line-height: 20px;
    font-size: 14px;
```
in order to match the design

File: extensions/blocks/social-previews/editor.scss [hunk 3]
```diff
@@ -12,3 +12,75 @@
 		fill: currentColor;
 	}
 }
+
+.jetpack-social-previews__upgrade-description {
+	margin-bottom: 1em;
+}
+
+.jetpack-social-previews__upgrade-heading {
+	font-size: 1.9em;
```
Comment (cpapazoglou) [inline extensions/blocks/social-previews/editor.scss:21]: Just for me learning up... How do you manage effortlessly using `em` when the design has `px` ❓

File: extensions/blocks/social-previews/upgrade.js [hunk 4]
```diff
@@ -1,6 +1,51 @@
...
+					</li>
+					<li>
+						{ __(
+							'Control how page titles will appear on Google search results and social networks.',
+							'jetpack'
+						) }
+					</li>
+					<li>
+						{ __(
+							'Customize your front page meta data to change hour your site appears to search engines.',
```
Comment (cpapazoglou) [inline extensions/blocks/social-previews/upgrade.js:41]: `hour` -> `how`

**Comment by marekhrabe** [comment]
Truly fullwidth image started pushing the text content much lower, I’ve decided to put a max-width on it so the text stays reasonably above the fold.

**Pull request merged**